### DISCUSSION TITLE
Compiling googletest for 32-bit target, embedded system

### DISCUSSION THREAD

**Discussion by DanBellmann** [google/googletest#4366] [Community Help]
Dear community members, 
I want to compile googletest with a 32-bit compiler without changing the source code of googletest. 
I use CMake to generate the build files. The compiler is not officialy supported by GoogleTest nor by CMake but still usable. 
When executing make in the googletest-build directory I get errors like "long long" is nonstandard. 

1.Question: Is there a 32-bit source code of googletest? The version has to be under 1.7.0. If not thats also okay!
2.Question: How do I avoid errors like these?

The GNU compiler gcc can do this with the option -m32, my compiler doesn't have such an option. 

Thank you for your help and time!

**Comment by pgawro** [comment]
@derekmauro what do you think it will be ok if I introduce change with stdint.h  and replace things like long long to uint64_t?

**Reply by derekmauro** [reply]
> @derekmauro what do you think it will be ok if I introduce change with stdint.h and replace things like long long to uint64_t?

It depends on the context. `int64_t` and `long long` do not have the same meaning.

`int64_t` is an alias type. `long long` is a built-in type.

Overload sets should to use `long long`, not `int64_t`, otherwise you risk defining overloads for the same underlying type twice, if aliases are setup that way.

This question is about GoogleTest <= 1.7.0, which is long out of date, and is also asking about an unsupported compiler. I'm not going to put any effort into supporting this use case.

**Reply by pgawro** [reply]
Ah, ok. Better ask than be sorry :)
